### PR TITLE
Gra 892 move proxy model to netmaker

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

**Review by abhishek9686 — COMMENTED** [review]
why were all models moved from proxy to server, there is only one or two structs that is being shared by client and server that needs to be moved

**Comment by mattkasun** [comment]
> 

It was easier to just move the entire package

**Review by Aceix — APPROVED** [review]

**Comment by abhishek9686** [comment]
> > 
> 
> It was easier to just move the entire package

this is will just cause unnecessary updates to server even for irrelevant changes when models being used only on client side

**Comment by theguy951357** [comment]
will wait on changes

File: nmproxy/models/models.go [hunk 1]
```diff
@@ -6,55 +6,29 @@ import (
...
-	// NoProxy - constant for no ProxyAction
-	NoProxy ProxyAction = "NO_PROXY"
 )
 
 // PeerConnMap - type for peer conn config map
 type PeerConnMap map[string]*Conn
 
-// Proxy - struct for proxy config
-type Proxy struct {
-	PeerPublicKey  wgtypes.Key
```
Comment (abhishek9686) [inline nmproxy/models/models.go:40]: I believe this only model is used only on netclient side, should be kept here

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: nmproxy/models/models.go (hunk 1 above)
Comment (mattkasun) [reply nmproxy/models/models.go]: moved back to netclient

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**